rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/integration/UserTest.java

```diff
@@ -54,20 +57,35 @@ public void before() {
         user.setFullname("Jane Doe");
         userList.add(user);
 
-		given(this.userServiceMock.getAllUsers()
-		).willReturn(userList);
-	}
-
-	@Test
-	public void should_get_all_users() throws IOException {
-		HttpHeaders headers = getHeaders(port);
-		ResponseEntity<String> response =
-				new TestRestTemplate().exchange("http://localhost:" + port + "/api/users",
-						HttpMethod.GET,
-						new HttpEntity<>(null, headers),
-						String.class);
+        given(this.userServiceMock.getAllUsers()).willReturn(userList);
+
+        given(this.userServiceMock.getUser(user.getId())).willReturn(user);
+    }
+
+    @Test
+    public void should_get_all_users() throws IOException {
+        ResponseEntity<String> response = sendRequest("http://localhost:" + port + "/api/users");
         assertThat(HttpStatus.OK, is(response.getStatusCode()));
 
-		TestHelper.checkResponse(response.getBody(), "users", 2);
-	}
+        TestHelper.checkResponse(response.getBody(), "users", 2);
+    }
+
+    @Test
+    public void should_get_single_user_with_correct_self_link() throws IOException {
```

Comment:
I would rather name this method `should_get_single_user_by_user_self_link`

Reply:
I updated the name to clarify the test intention.